pkg/sql/parser/sql.y

```diff
@@ -4574,10 +4583,116 @@ create_stmt:
 | create_extension_stmt  // EXTEND WITH HELP: CREATE EXTENSION
 | create_external_connection_stmt // EXTEND WITH HELP: CREATE EXTERNAL CONNECTION
 | create_virtual_cluster_stmt     // EXTEND WITH HELP: CREATE VIRTUAL CLUSTER
+| create_logical_replication_stream_stmt     // EXTEND WITH HELP: CREATE LOGICAL REPLICATION STREAM
 | create_schedule_stmt   // help texts in sub-rule
 | create_unsupported     {}
 | CREATE error           // SHOW HELP: CREATE
 
+// %Help: CREATE LOGICAL REPLICATION STREAM - create a new logical replication stream
+// %Category: Experimental
+// %Text:
+// CREATE LOGICAL REPLICATION STREAM 
+//  FROM 'stream_uri' 
+//  ON <TABLE remote_name | TABLES (remote_name, ...) | DATABASE remote_name>
+//  INTO <TABLE remote_name | TABLES (remote_name, ...) | DATABASE remote_name>
+//  [WITH
+//  < MODE= immediate | transactional > |
+//  < CURSOR = start_time > |
+//  < FUNCTION 'udf' FOR TABLE local_name  , ... >
+// ]
+create_logical_replication_stream_stmt:
+  CREATE LOGICAL REPLICATION STREAM FROM d_expr ON logical_replication_resources INTO logical_replication_resources opt_logical_replication_options
+  {
+    /* SKIP DOC */
+    $$.val = &tree.CreateLogicalReplicationStream{
+      PGURL: $6.expr(),
+      On: $8.logicalReplicationResources(),
+      Into: $10.logicalReplicationResources(),
+      Options: *$11.logicalReplicationOptions(),
+    }
+  }
+| CREATE LOGICAL REPLICATION STREAM error // SHOW HELP: CREATE LOGICAL REPLICATION STREAM
+
+logical_replication_resources:
+  TABLE logical_replication_resources_list
```

Comment:
this one should just be `db_object_name` right?

Reply:
I ended up just using the logical_replication_resources_list first case which is just db_object_name. Since the block was just repeated basically. 

Can revert it back though for clarity.

Reply:
I think it is more than just for clarity: this production means `TABLE x, y, z` is matched as well, not just `TABLE x` so we're back to the case where if we ever want a mixture of target productions it gets messy

Reply:
Oh yes I see what youre saying. I've added an error test case for this too in the parser test to make sure it doesn't parse.